core/src/main/scala/kafka/log/ProducerIdMapping.scala

```diff
@@ -282,8 +266,7 @@ class ProducerIdMapping(val config: LogConfig,
 
   def truncateAndReload(logEndOffset: Long, currentTime: Long) {
     truncateSnapshotFiles(logEndOffset)
-    def checkNotExpired = (producerIdEntry: ProducerIdEntry) => { isEntryValid(currentTime, producerIdEntry) }
-    loadFromSnapshot(logEndOffset, checkNotExpired)
+    loadFromSnapshot(logEndOffset, producerIdEntry => isEntryValid(currentTime, producerIdEntry))
```

Comment:
Should we remove any snapshot whose offset is > logEndOffset? Should we also just call updateMapEndOffset() here instead of letting the caller do it?